### PR TITLE
Implement image manager

### PR DESCRIPTION
**Pull request by Imperat**
Do not merge it! Work in progress!

### CODE REVIEW

File: app/neural/network.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 from pybrain.tools.customxml import NetworkWriter, NetworkReader
 from pybrain.supervised.trainers import BackpropTrainer
 from pybrain.datasets import SupervisedDataSet
-from functools import reduce
```
Comment (Imperat) [inline app/neural/network.py:7]: this was unused import

File: app/managers/img_manager.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    with TemporaryDirectory() as td:
+        zip_file.extractall(td)
+        for file_name in listdir(td):
+            _read_image(path.join(td, file_name))
+
+
+def _read_image(file_name, digit=None):
+    if digit is None:
+        if '_' not in file_name:
+            # Invalid file name in Archive. May be log about it?
```
Comment (Imperat) [inline app/managers/img_manager.py]: May be we need to create some python module with application Exceptions and try in this place for example "InvalidArchivedImageName" exception?

File: app/managers/img_manager.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+def _read_archive(data):
+    archive = BytesIO(data)
+    zip_file = ZipFile(archive, "r")
+    with TemporaryDirectory() as td:
+        zip_file.extractall(td)
+        for file_name in listdir(td):
+            _read_image(path.join(td, file_name))
+
+
+def _read_image(file_name, digit=None):
```
Comment (Imperat) [inline app/managers/img_manager.py]: I add option "digit" for future. If we need to implement loading images as single files, we need to call _read_image method with taken digit from user

File: ui/views.py [hunk 4]
```diff
@@ -151,7 +151,7 @@ def add_sample():
         r = requests.post(api_url + '/neural/sample/add',
                           headers=headers,
                           auth=(session['user']['token'], 'unused'),
-                          files=files)
+                          data=form.archived_images.data)
```
Comment (wlad031) [inline ui/views.py:150]: ```files``` variable is unused now, i think, you may remove it
And in ```recognize``` function you need to do the same change

**Review by wlad031 — CHANGES_REQUESTED** [review]

File: ui/views.py (hunk 4 above)
Comment (Imperat) [reply ui/views.py:150]: Done

**Review by wlad031 — APPROVED** [review]

**Pull request merged**